File: Pyto/Python Bridging/PyMainThread.m
//
//  NSObject+PyMainThread.m
//  Pyto
//
//  Created by Adrian Labbé on 1/24/19.
//  Copyright © 2019 Adrian Labbé. All rights reserved.
//

#import <Foundation/Foundation.h>

@interface PyMainThread: NSObject
    
+ (void) runAsync:(void (^)(void))block;
+ (void) runSync:(void (^)(void))block;
@end

@implementation PyMainThread
+ (void) runAsync:(void (^)(void))block {
    dispatch_async(dispatch_get_main_queue(), block);
}

+ (void) runSync:(void (^)(void))block {
    dispatch_sync(dispatch_get_main_queue(), block);
}
@end

File: Pyto/Python Bridging/PyMainThread.swift
//
//  PyMainThread.swift
//  Pyto
//
//  Created by Adrian Labbe on 9/15/18.
//  Copyright © 2018 Adrian Labbé. All rights reserved.
//

import Foundation

/// A class representing the main thread for the Python API.
@objc class PyMainThread: NSObject {
    
    /// Execute the given code async on the main thread.
    @objc static func runAsync(_ code: @escaping (() -> Void)) {
        DispatchQueue.main.async {
            code()
        }
    }
    
    /// Execute the given code sync on the main thread.
    @objc static func runSync(_ code: @escaping (() -> Void)) {
        DispatchQueue.main.sync {
            code()
        }
    }
}
